alica_tests/src/test/test_alica_scheduling.cpp

```diff
@@ -1,256 +1,128 @@
 #include "test_alica.h"
-#include <alica/test/CounterClass.h>
-#include <alica_tests/behaviours/BehAAA.h>
-#include <alica_tests/behaviours/BehBAA.h>
 
 #include <alica/test/Util.h>
-#include <alica_tests/test_sched_world_model.h>
 #include <gtest/gtest.h>
 
-namespace alica
+namespace alica::test
 {
-namespace
-{
-
-class AlicaSchedulingPlan : public AlicaSchedulingTestFixture
-{
-protected:
-    const char* getMasterPlanName() const override { return "SchedulingTestMasterPlan"; }
-    bool stepEngine() const override { return true; }
-};
 
-TEST_F(AlicaSchedulingPlan, scheduling)
+TEST_F(SingleAgentTestFixture, transitionAfterInitTest)
 {
-    ae->start();
-    CounterClass::called = 0;
+    // Checks if outgoing transitions are evaluated after all plans and behaviours in a state have been initialized
 
-    STEP_UNTIL(CounterClass::called == 1);
-    // init of scheduling plan 1
-    ASSERT_EQ(1, CounterClass::called);
-    CounterClass::called += 1; // allow transition
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "SchedulingTestState")) << _tc->getLastFailure();
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->getActivePlan("SchedulingTestPlan"));
+    ASSERT_TRUE(_tc->setTransitionCond("SchedulingTestPlan", "ChooseTestState", "TransitionAfterInitTestState")) << _tc->getLastFailure();
 
-    STEP_UNTIL(CounterClass::called == 4);
-    // init of scheduling plan 2 and 3
-    ASSERT_EQ(4, CounterClass::called);
-    CounterClass::called += 1; // allow transition
-
-    STEP_UNTIL(CounterClass::called == 7);
-    // onTermination of scheduling plan 2 and 3 called
-    ASSERT_EQ(7, CounterClass::called);
-    CounterClass::called += 1; // allow transition
-
-    STEP_UNTIL(CounterClass::called == 9);
-    // onTermination of scheduling plan 1 called
-    ASSERT_EQ(9, CounterClass::called);
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->getActivePlan("TransitionAfterInitTestPlan"));
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->getActivePlan("SchedulingTestPlan"));
```

Comment:
Just checking if `SchedulingTestPlan` is active is not of much use because this plan is the parent plan & therefore will be active anyway. Instead you also need to check if `SchedulingTestPlan` has reached the success state i.e use `isSuccess(SchedulingTestPlan)`